sdk/modelsrepository/Azure.Iot.ModelsRepository/src/Fetchers/RemoteModelFetcher.cs

```diff
@@ -40,50 +131,88 @@ public async Task<FetchResult> FetchAsync(string dtmi, Uri repositoryUri, Cancel
-                    };
+                    return GetContent(message.Response.ContentStream);
+                }
+                else
+                {
+                    throw _clientDiagnostics.CreateRequestFailedException(message.Response);
                 }
-
-                ResolverEventSource.Shared.ErrorFetchingModelContent(tryContentPath);
-                remoteFetchError = string.Format(CultureInfo.CurrentCulture, StandardStrings.ErrorFetchingModelContent, tryContentPath);
             }
-
-            throw new RequestFailedException(remoteFetchError);
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
         }
 
-        private static string GetPath(string dtmi, Uri repositoryUri, bool expanded = false)
+        private async Task<string> EvaluatePathAsync(string path, CancellationToken cancellationToken = default)
         {
-            string absoluteUri = repositoryUri.AbsoluteUri;
-            return DtmiConventions.DtmiToQualifiedPath(dtmi, absoluteUri, expanded);
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope("RemoteModelFetcher.EvaluatePath");
+            scope.Start();
+
+            try
+            {
+                using HttpMessage message = CreateGetRequest(path);
+
+                await _pipeline.SendAsync(message, cancellationToken).ConfigureAwait(false);
+
+                if (message.Response.Status >= 200 && message.Response.Status <= 299)
+                {
+                    return await GetContentAsync(message.Response.ContentStream, cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+                    throw await _clientDiagnostics.CreateRequestFailedExceptionAsync(message.Response).ConfigureAwait(false);
+                }
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
         }
 
-        private async Task<string> EvaluatePathAsync(string path, CancellationToken cancellationToken)
+        private HttpMessage CreateGetRequest(string path)
         {
-            Request request = _pipeline.CreateRequest();
+            HttpMessage message = _pipeline.CreateMessage();
+            Request request = message.Request;
             request.Method = RequestMethod.Get;
-            request.Uri = new RequestUriBuilder();
-            request.Uri.Reset(new Uri(path));
+            var uri = new RequestUriBuilder();
+            uri.Reset(new Uri(path));
+            request.Uri = uri;
 
-            Response response = await _pipeline.SendRequestAsync(request, cancellationToken).ConfigureAwait(false);
+            return message;
+        }
 
-            if (response.Status >= 200 && response.Status <= 299)
+        private static string GetContent(Stream content)
+        {
+            using (JsonDocument json = JsonDocument.Parse(content))
```

Comment:
Want to set project lang to 8, so we can use the simplified `using` statement?

Reply:
We can, I changed it.